runtime/src/append_vec.rs

```diff
@@ -138,6 +143,35 @@ impl AppendVec {
         }
     }
 
+    #[allow(clippy::mutex_atomic)]
+    pub fn new_empty_map(current_len: usize) -> Self {
+        let map = MmapMut::map_anon(1).expect("failed to map the data file");
```

Comment:
This error handling is made more critical rather undesirably [from `Err`](https://github.com/solana-labs/solana/pull/7373/files#diff-657d1e7648a9fb74cb1582a9c48beb4bL419) to just move the code here. But combined with [this consideration](https://github.com/solana-labs/solana/pull/7373/files#r355330172), this can be tolerated to avoid large refactoring. :) And this mmap will almost never fail.